src/v/datalake/translation/deps.cc

```diff
@@ -577,4 +590,59 @@ translation_context::make_default_translation_context(
       std::move(probe));
 }
 
+// note: this is a temporary implementation to get the code compiling
+// to be filled with a proper implementation that tracks windows based
+// on batch timestamps + heuristics
+// Even the interface is temporary and subject to change depending on
+// the changes needed to track windows.
+class partition_lag_tracker : public translation_lag_tracker {
+public:
+    explicit partition_lag_tracker(
+      ss::lw_shared_ptr<cluster::partition> partition,
+      cluster::topic_table& topics)
+      : _partition(std::move(partition))
+      , _topics(topics)
+      , _current_window_end(scheduling::clock::now()) {
+        reset_to_next_window();
+    }
+
+    bool should_finish_inflight_translation() final {
+        auto window_ended = scheduling::clock::now() >= _current_window_end;
+        if (window_ended) {
+            reset_to_next_window();
+        }
```

Comment:
Just noting that at first glance bit surprising for this to be resetting the window too. I would've expected callers to need to reset this explicitly, e.g. when successfully checkpointing translation success or something

Reply:
>the way we calculate lag with the heuristic and timestamp approximation today, the window is automatically reset after checkpointing state

Yea I can see that, I don't love that the interface reflects the implementation detail. Not too important though, I guess we can change it as the heuristic evolves.